website/docs/language/resources/ephemeral.mdx

```diff
@@ -0,0 +1,137 @@
+optionally renew it one or more times, and then close it.
+
+1. When Terraform needs to access the ephemeral resource result, it will open
+the ephemeral resource. For example, opening a vault secret ephemeral resource
+means the Vault provider will obtain the lease and return the secret.
+
+1. If Terraform needs access to the ephemeral resource for longer than the
+expiration time enforced by the remote system, Terraform will ask the provider
+to renew it periodically. For example, renewing a Vault secret ephemeral
+resource means the Vault provider will call Vault's lease renewal API endpoint
+to extend the expiration time.
+
+1. Once Terraform no longer needs the ephemeral resource, Terraform will close
+it. This will happen once any providers that depend on the ephemeral resource
+have completed all of their work for the current phase. For example, closing a
+Vault secret ephemeral resource means the Vault provider will explicitly end the
+lease, allowing Vault to immediately revoke the associated credentials.
+
+Terraform will follow these lifecycle steps for each instance of an ephemeral
+resource in a given configuration.
+
+### Dependency graph
+
+Ephemeeral resources form nodes in Terraform's dependency graph which interact
+similarly to resources and data sources. For example, when a resource or data
+source depends on an attribute of an ephemeral resource, Terraform will
+automatically provision the ephemeral resource first.
+
+## Configuration model
+
+The following list outlines field hierarchy, language-specific data types, and
+requirements in the `moved` block. 
+
+- [`ephemeral`](#ephemeral): map
+  - [`attributes`](#ephemeral): string
+  - [`meta-arguments`](#ephemeral): string
+
+## Complete configuration
+
+An `ephemeral` block has the following form:
+
+```hcl
+ephemeral "<resource_type>" "<resource_name>" {
+  <attributes>
+  <meta-arguments>
+}
+```
+
+## Specification
+
+This section provides details about the fields you can configure in the `ephemeral` block.
+
+### `ephemeral`
+
+Map that specifies addresses for the resource. The following table describes the fields you can set in the `moved` block.
+
+<!--FIXME: Not sure how to format/address this for this type of block -->
+
+| Field | Description | Type | Required |
+| ---   | ---         | ---  | ---      |
+| `from` | Specifies a resource's previous address. The syntax allows Terraform to select modules, resources, and resources inside child modules. | string | required |
+| `to` | Specifies the new address to relocate the resource to. The syntax allows Terraform to select modules, resources, and resources inside child modules. | string | required |
+
+## Example
+
+The following example uses the `random` provider to provision an ephemeral
+resource representing a password:
+
+```hcl
+provider random {}
+
+ephemeral "random_password" "password" {
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+```
```

Comment:
It'd also be super nice to have a more complicated example, like setting up a provider configuration with an ephemeral resource, [like in this demo](https://hashicorp.slack.com/archives/C071HC4JJCC/p1729859992489559). If it wouldn't be too much work I think it'd really be the bread and butter of this article if you can pull it off